### PR TITLE
Doc consistency sweep after bug-killing campaign (ROADMAP / HISTORY / SKILL.md)

### PR DESCRIPTION
**Pull request by aallan**
## Summary

Doc-layer consistency sweep after the bug-killing campaign closed in v0.0.134.  Three focused commits:

1. **ROADMAP rework** — replace `What's next — short-term priorities` (anchored entirely in the now-closed campaign, with a one-row priority table after it) with `What's next — agent-integration push`.  Three priorities pulled from later-milestone phases that share the property of being independent of compiler internals: **#222 LSP**, **#523 `vera context`**, **#370 `Inference.complete` max_tokens / temperature**.
2. **HISTORY compaction** — eight entries in the Stage 11 release table (v0.0.119-v0.0.134) were 2-3× the density of Stage 5-9 entries, with implementation details and parenthetical asides that don't match the established style.  Compacted to "feature/fix description + issue link, optional one-clause qualifier".  Long form stays in CHANGELOG.
3. **SKILL.md cleanup** — drop three "Known Bugs" rows for issues closed in the campaign (#475, #487, #535) and rewrite the closure-capture subsection that was telling agents to use a now-obsolete "lift to a helper" workaround for pair-type captures (#535 closed in v0.0.130).  Add the still-open #568 (url_parse leading-colon drop) for parity with KNOWN_ISSUES.md.

## Why this matters

The user-facing KNOWN_ISSUES.md has stayed clean throughout the campaign because it's part of every release-doc sync.  The agent-facing layer (SKILL.md) accumulated stale "Known Bugs" rows silently because no validation gate watches it line-by-line.  Agents writing Vera against this doc were generating convoluted code to work around bugs that no longer exist.

Net delta: **−108 lines** across the three commits.  Net signal-to-noise improvement is much larger because the deleted lines were specifically the ones leading agents wrong.

## Test plan

- [x] `pytest tests/ -q` — 3,702 passed, 14 skipped
- [x] `python scripts/check_doc_counts.py` — consistent
- [x] `python scripts/check_limitations_sync.py` — 27 KNOWN_ISSUES / 15 vera/README / 2 spec, consistent
- [x] `python scripts/check_skill_examples.py` — 55 Vera blocks parse, 62 allowlisted, 0 failures
- [x] `python scripts/build_site.py && python scripts/check_site_assets.py` — site assets up-to-date (`docs/SKILL.md` + `docs/llms-full.txt` regenerated as part of commit 3)
- [x] `python scripts/fix_allowlists.py` — all up to date (line-shift fix included in commit 3)
- [x] All pre-commit hooks pass on each of the three commits

## Notes for reviewers

**This is doc-only.**  No code changes, no behavior changes, no version bump.  After merge, retag `v0.0.134` to HEAD so the released artefact reflects the cleaned SKILL.md (the current `v0.0.134` tag points at a commit where SKILL.md still listed three closed issues as Known Bugs and had a stale closure-capture workaround).

**Companion PR**: #580 (already open) is a tiny hotfix for a stale visible version on `docs/index.html` plus a hardening of `check_version_sync.py`.  Independent of this sweep; can merge in either order.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Expanded HISTORY Stage 11 entries (v0.0.119–v0.0.134) with detailed feature notes
  * Replaced short-term priorities with an agent-integration focused ROADMAP and clarified implementation ordering
  * Updated SKILL guidance to broaden closure-capture semantics, add pair-type and nested-closure examples
  * Added Unreleased and 0.0.134 changelog notes summarising the documentation sweep

* **Tests**
  * Expanded validation allowlist to cover many additional SKILL examples and edge cases

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -253,22 +253,22 @@ Stage 11 shifts focus from evaluation infrastructure to the standard library and
...
+| v0.0.131 | 5 May | **GC infrastructure batch — closes [#487](https://github.com/aallan/vera/issues/487) and [#348](https://github.com/aallan/vera/issues/348)** (`$alloc` multi-page grow, worklist size + overflow trap). |
+| v0.0.132 | 5 May | **Opaque-handle GC-rooting hygiene — closes [#347](https://github.com/aallan/vera/issues/347) and [#490](https://github.com/aallan/vera/issues/490)**. |
+| v0.0.133 | 5 May | **Iterative array builders no longer leak closure return-value root — closes [#570](https://github.com/aallan/vera/issues/570)** (`array_map` / `_mapi` / `_fold` / `_sort_by`). |
+| v0.0.134 | 6 May | **Active reclamation of host-store handles via heap-wrap-as-ADT — closes [#573](https://github.com/aallan/vera/issues/573), [#575](https://github.com/aallan/vera/issues/575), [#576](https://github.com/aallan/vera/issues/576), [#579](https://github.com/aallan/vera/issues/579)** (Map / Set / Decimal). |
```
[the remark this answers is not in the document]
Comment (aallan) [reply HISTORY.md:271]: @coderabbitai Skip — finding doesn't apply here.

The reviewer's framing assumes the standard "no-version-bump housekeeping PR shouldn't touch the released-version ledger" rule, which is correct in the general case but doesn't fit this PR.

**The HISTORY compaction is the deliberate point of this PR**, not collateral damage from it. The Stage 11 release table had drifted to 2–3× the density of Stage 5–9 entries (implementation details, parenthetical asides, metric examples) over the bug-killing campaign. Restoring the established density is the cleanup we set out to do; reverting the HISTORY edits would defeat the PR's stated purpose.

**The PR is also not no-version-bump — it lands in v0.0.134 via tag-move-to-HEAD.** The current `v0.0.134` tag points at a commit where SKILL.md still listed three closed issues as Known Bugs and HISTORY.md had bloated entries. After this PR merges the tag will be moved forward so the released artefact reflects the cleaned docs. So:

- Editing HISTORY's `v0.0.134` row (adding `#579` to the closes list, compacting the prose) is editing the row for the release this PR ships under.
- Compacting v0.0.119–v0.0.133 rows is part of the same sweep — they were the same campaign.
- The CHANGELOG entry has been moved from `[Unreleased]` to `[0.0.134]` for the same reason (commit `2591fb2`, with `Skip-changelog:` trailer to satisfy the CI gate which assumes the standard version-bump workflow).

So the "record this in CHANGELOG / ROADMAP / a GH issue per project policy" recommendation already applies on the CHANGELOG side, but the HISTORY content itself is the work product, not housekeeping prose about the work.
